spec/functional/command/repo/push_spec.rb

```diff
@@ -38,6 +44,16 @@ module Pod
       end
     end
 
+    it 'finds JSON podspecs' do
+      repo_make('test_repo')
+
+      Dir.chdir(temporary_directory) do
+        File.open('JSON.podspec.json',  'w') { |f| f.write('{}') }
+        cmd = command('repo', 'push', 'test_repo')
+        cmd.send(:podspec_files).should == [Pathname('JSON.podspec.json')]
```

Comment:
If anyone has any suggestions on how to easily test this without accessing the private method I'm open for ideas.